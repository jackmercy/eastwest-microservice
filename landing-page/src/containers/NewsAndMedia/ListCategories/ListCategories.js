/* eslint-disable no-unused-vars */
import Link from 'next/link';
import { useRouter } from 'next/router';
import { memo, useCallback, useEffect, useMemo, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import classNames from 'classnames';
import { useMediaQuery } from 'react-responsive';

import { InputSearch } from '../../../components';
import { useOnClickOutside } from '../../../hooks';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

import useStyles from './listCategories.styles';

const ListCategories = ({ data, onSearch, className }) => {
  const intl = useIntl();
  const router = useRouter();
  const moreRef = useRef();
  const classes = useStyles();
  const isMobile = useMediaQuery({
    query: `(max-width: ${BREAKPOINTS.maxMd}px)`,
  });
  const [activeCategory, setActiveCategory] = useState(0);
  const [openPopup, setOpenPopup] = useState(false);
  const categoriesDisplayed = useMemo(() => (isMobile ? 100 : 5), [isMobile]);
  const listCategories = useMemo(
    () => [
      { id: '', name: intl.formatMessage({ id: 'NEWS.CATEGORY.LATEST' }) },
      ...data,
    ],
    [intl],
  );
  const moreCategories = useMemo(
    () => listCategories.slice(categoriesDisplayed),
    [listCategories, categoriesDisplayed],
  );

  useEffect(() => {
    const index = data.findIndex((c) => c.slug === router.query?.category);
    setActiveCategory(index + 1);
  }, [router, data]);

  const handleMoreClick = useCallback(() => {
    setOpenPopup(!openPopup);
  }, [openPopup]);

  useOnClickOutside(moreRef, () => {
    setOpenPopup(false);
  });

  const renderSlug = useCallback(
    (slug) => (slug ? `/news-and-media/${slug}` : '/news-and-media'),
    [],
  );

  return (
    <div id="list-categories" className={classNames(classes.root, className)}>
      <ul
        className={classNames('hide-scrollbar', classes.list)}
        data-test="list-categories-list"
      >
        {listCategories.slice(0, categoriesDisplayed).map((item, index) => (
          <li
            key={item.id}
            className={classNames(
              classes.listItem,
              activeCategory === index && 'active',
            )}
          >
            <Link href={renderSlug(item.slug)} passHref>
              <a href={renderSlug(item.slug)} className={classes.link}>
                {item.name}
              </a>
            </Link>
          </li>
        ))}
        {!isMobile && moreCategories.length > 0 ? (
          <li
            ref={moreRef}
            role="presentation"
            className={classNames(
              classes.listItem,
              classes.moreItems,
              openPopup && 'open',
              activeCategory >= categoriesDisplayed && 'active',
            )}
            onClick={handleMoreClick}
          >
            {intl.formatMessage({ id: 'NEWS.CATEGORY.MORE' })}
            <i className="icon-triangle-down" />
            {openPopup ? (
              <ul className={classNames('custom-scrollbar', classes.popupList)}>
                {moreCategories.map((item, index) => (
                  <li
                    key={item.id}
                    className={classNames(
                      classes.popupListItem,
                      activeCategory === categoriesDisplayed + index &&
                        'active',
                    )}
                  >
                    <Link href={renderSlug(item.slug)} passHref>
                      <a href={renderSlug(item.slug)} className={classes.link}>
                        {item.name}
                      </a>
                    </Link>
                  </li>
                ))}
              </ul>
            ) : null}
          </li>
        ) : null}
      </ul>
      {/* MVP hidden */}
      {/* <InputSearch
        size="small"
        onSearch={onSearch}
        className={classes.inputSearch}
        inputClassName={classes.inputClassName}
        placeholder={intl.formatMessage({ id: 'NEWS.SEARCH' })}
        dataTest="list-categories-input-search"
      /> */}
    </div>
  );
};
ListCategories.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
    }),
  ),
  onSearch: PropTypes.func.isRequired,
  className: PropTypes.string,
};
ListCategories.defaultProps = {
  data: [],
  className: '',
};

export default memo(ListCategories);
