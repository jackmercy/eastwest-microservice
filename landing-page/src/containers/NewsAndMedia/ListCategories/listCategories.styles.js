import { createUseStyles } from 'react-jss';

import { hexToRgbA, rem } from '../../../utils/jss';
import COLORS from '../../../constants/COLORS';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  list: {
    display: 'flex',
    listStyle: 'none',
    margin: 0,
    padding: 0,
  },
  listItem: {
    padding: rem(8),
    cursor: 'pointer',
    fontWeight: 'bold',
    fontSize: rem(18),
    lineHeight: rem(24),
    position: 'relative',
    whiteSpace: 'nowrap',
    '& + $listItem': { marginLeft: rem(32) },
    '&::after': {
      content: '""',
      position: 'absolute',
      width: 0,
      left: 0,
      bottom: 0,
      backgroundColor: COLORS.primary,
      transition: 'width .5s ease',
      borderRadius: rem(4),
      height: rem(2),
    },
    '&:hover': {
      '& > $link': { color: COLORS.primary },
      '&::after': { width: '100%' },
    },
    '&.active::after': { width: '100%' },
  },
  link: { color: COLORS.textBlack01, textDecoration: 'none' },
  moreItems: {
    display: 'flex',
    alignItems: 'center',
    '& > [class^=icon-]': {
      marginLeft: rem(10),
      fontSize: rem(8),
      transition: 'transform .5s ease',
    },
    '&.open': {
      color: COLORS.primary,
      '&::after': { width: '100%' },
      '& > [class^=icon-]': { transform: 'rotate(-180deg)' },
    },
  },
  popupList: {
    zIndex: 20,
    display: 'block',
    position: 'absolute',
    listStyle: 'none',
    margin: 0,
    left: 0,
    top: rem(48),
    backgroundColor: COLORS.bgWhite,
    borderRadius: rem(8),
    padding: [rem(8), 0],
    minWidth: rem(220),
    maxWidth: rem(300),
    maxHeight: rem(250),
    overflow: 'auto',
    boxShadow: '2px 4px 25px rgba(0, 0, 0, 0.25)',
  },
  popupListItem: {
    padding: [rem(8), rem(24)],
    cursor: 'pointer',
    fontWeight: 'bold',
    fontSize: rem(18),
    lineHeight: rem(24),
    color: COLORS.textBlack01,
    '&:hover, &.active': {
      backgroundColor: COLORS.bgGray,
    },
  },
  inputSearch: { width: '100%', maxWidth: rem(376), marginLeft: rem(24) },
  inputClassName: {
    borderRadius: rem(100),
    '&:not(:focus)': {
      border: 'none',
      color: COLORS.textGray02,
      backgroundColor: hexToRgbA(COLORS.bgBlack01, 0.05),
      '& + [class^=icon-]': { color: COLORS.textGray02 },
    },
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxMd}px)`]: {
    root: { display: 'block', overflow: 'hidden' },
    list: { overflow: 'auto' },
    inputSearch: { marginLeft: 0, marginTop: rem(24), maxWidth: '100%' },
  },
});
