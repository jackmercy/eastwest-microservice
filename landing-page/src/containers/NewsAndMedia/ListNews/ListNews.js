import { useRouter } from 'next/router';
import { memo, useCallback, useMemo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { useIntl } from 'react-intl';
import ReactPaginate from 'react-paginate';

import Item from './Item';
import useStyles from './listNews.styles';

const ITEM_PER_PAGE = 5;

const ListNews = ({ data, categories, totalNews, className }) => {
  const intl = useIntl();
  const classes = useStyles();
  const router = useRouter();
  const currentPage = useMemo(() => router.query?.page - 1 || 0, [router]);

  const handlePageChange = useCallback(
    ({ selected }) => {
      router.push({
        pathname: window.location.pathname,
        query: { page: selected + 1 },
      });
    },
    [router],
  );

  return data && data.length > 0 ? (
    <div className={classNames(classes.root, className)}>
      {data.slice(0, ITEM_PER_PAGE).map((item) => (
        <Item
          key={item.id}
          data={{
            ...item,
            category: categories.find((c) => c.id === item.category),
          }}
          dataTest="list-news-item"
        />
      ))}
      {totalNews > ITEM_PER_PAGE ? (
        <ReactPaginate
          previousLabel={intl.formatMessage({ id: 'NEWS.PAGINATION.BACK' })}
          nextLabel={intl.formatMessage({ id: 'NEWS.PAGINATION.NEXT' })}
          breakLabel="..."
          pageCount={Math.ceil(totalNews / ITEM_PER_PAGE)}
          marginPagesDisplayed={1}
          pageRangeDisplayed={2}
          forcePage={currentPage}
          onPageChange={handlePageChange}
          containerClassName={classes.paginationContainer}
          activeLinkClassName={classes.paginationActiveLink}
          pageClassName={classes.paginationButton}
          previousClassName={classes.paginationButton}
          nextClassName={classes.paginationButton}
          previousLinkClassName={classes.paginationLink}
          nextLinkClassName={classes.paginationLink}
          pageLinkClassName={classes.paginationLink}
        />
      ) : null}
    </div>
  ) : (
    <p className={classNames(classes.empty, className)}>
      {intl.formatMessage({ id: 'NEWS.EMPTY' })}
    </p>
  );
};
ListNews.propTypes = {
  totalNews: PropTypes.number.isRequired,
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      category: PropTypes.string.isRequired,
      thumbnail: PropTypes.string.isRequired,
    }),
  ),
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
    }),
  ),
  className: PropTypes.string,
};
ListNews.defaultProps = {
  data: [],
  categories: [],
  className: '',
};

export default memo(ListNews);
