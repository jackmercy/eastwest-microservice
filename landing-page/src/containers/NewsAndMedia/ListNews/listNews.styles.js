import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';
import COLORS from '../../../constants/COLORS';

export default createUseStyles({
  root: {},
  paginationContainer: {
    listStyle: 'none',
    margin: [rem(80), 0, 0],
    padding: 0,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  paginationButton: { margin: [0, rem(4)] },
  paginationLink: {
    fontWeight: 500,
    fontSize: rem(16),
    lineHeight: rem(24),
    cursor: 'pointer',
    padding: [0, rem(8)],
    display: 'block',
    borderRadius: rem(2),
    color: COLORS.textBlack01,
    textDecoration: 'none',
    border: `1px solid ${COLORS.primary}`,
    '&:hover': {
      fontWeight: 600,
      color: COLORS.textWhite,
      backgroundColor: COLORS.primary,
    },
  },
  paginationActiveLink: {
    fontWeight: 600,
    color: COLORS.textWhite,
    backgroundColor: COLORS.primary,
  },
  empty: { textAlign: 'center', fontSize: rem(16), lineHeight: rem(24) },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    pagination: { marginTop: rem(32) },
  },
});
