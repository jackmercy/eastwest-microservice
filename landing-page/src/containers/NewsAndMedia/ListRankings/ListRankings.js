import { memo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Item from './Item';
import useStyles from './listRankings.styles';

const ListRankings = ({ data, categories, className }) => {
  const classes = useStyles();

  return data && data.length > 0 ? (
    <div className={classNames(classes.root, className)}>
      {data.slice(0, 5).map((item, index) => (
        <Item
          key={item.id}
          index={index + 1}
          data={{
            ...item,
            category: categories.find((c) => c.id === item.category),
          }}
        />
      ))}
    </div>
  ) : null;
};
ListRankings.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      category: PropTypes.string.isRequired,
      thumbnail: PropTypes.string.isRequired,
    }),
  ),
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
    }),
  ),
  className: PropTypes.string,
};
ListRankings.defaultProps = {
  data: [],
  categories: [],
  className: '',
};

export default memo(ListRankings);
