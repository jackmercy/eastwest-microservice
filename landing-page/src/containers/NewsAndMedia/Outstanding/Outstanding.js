import { memo } from 'react';
import PropTypes from 'prop-types';

import { Content } from '../../components';

import Item from './Item';
import useStyles from './outstanding.styles';

const Outstanding = ({ data, categories }) => {
  const classes = useStyles();

  return data && data.length > 0 ? (
    <Content className={classes.root}>
      <div className={classes.main}>
        <Item
          data={{
            ...data[0],
            category: categories.find((c) => c.id === data[0]?.category),
          }}
          main
          className={classes.mainItem}
          dataTest="outstanding-main-item"
        />
      </div>
      <div className={classes.others}>
        {data.slice(1, 3).map((item) => (
          <Item
            key={item.id}
            data={{
              ...item,
              category: categories.find((c) => c.id === item.category),
            }}
            className={classes.otherItem}
            size="small"
            dataTest="outstanding-sub-item"
          />
        ))}
      </div>
    </Content>
  ) : null;
};
Outstanding.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      category: PropTypes.string.isRequired,
      thumbnail: PropTypes.string.isRequired,
    }),
  ),
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
    }),
  ),
};
Outstanding.defaultProps = {
  data: [],
  categories: [],
};

export default memo(Outstanding);
