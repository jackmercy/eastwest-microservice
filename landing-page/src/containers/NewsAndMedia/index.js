import dynamic from 'next/dynamic';

export default dynamic(() => import('./NewsAndMedia'), { ssr: false });
