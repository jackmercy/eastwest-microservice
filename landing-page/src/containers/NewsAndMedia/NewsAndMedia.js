import { useRouter } from 'next/router';
import { memo, useCallback, useEffect, useMemo } from 'react';
import { useIntl } from 'react-intl';
import PropTypes from 'prop-types';

import { SEO } from '../../components';
import { scrollToElement } from '../../utils';
import { DEFAULT_SEO, LANDING_PAGE_DOMAIN } from '../../../configs';
import { Layout, Header, Footer, Content } from '../components';

import Outstanding from './Outstanding';
import ListNews from './ListNews';
import ListRankings from './ListRankings';
import ListCategories from './ListCategories';
import useStyles from './newsAndMedia.styles';

const NewsAndMedia = ({
  totalNews,
  categories,
  topNews,
  newsByCategory,
  newsRankings,
}) => {
  const intl = useIntl();
  const router = useRouter();
  const classes = useStyles();
  const seo = useMemo(() => {
    const category = categories.find((c) => c.slug === router.query?.category);
    if (category) {
      return {
        title: `${DEFAULT_SEO.PAGES.NEWS_AND_MEDIA.title} - ${category.name}`,
        description: category.shortDescription,
        url: `${LANDING_PAGE_DOMAIN}/${router.asPath}`,
        keywords: category.tags.join(', '),
      };
    }

    return DEFAULT_SEO.PAGES.NEWS_AND_MEDIA;
  }, [router, categories]);

  useEffect(() => {
    if (router.query?.category) {
      setTimeout(() => {
        scrollToElement('list-categories');
      }, 200);
    }
  }, [router.query]);

  const handleSearch = useCallback(() => {}, []);

  return (
    <Layout>
      <SEO {...seo} />
      <Header />
      <Outstanding data={topNews} categories={categories} />
      <Content className={classes.content}>
        <h2 className={classes.title}>
          {intl.formatMessage({ id: 'NEWS.TITLE' })}
        </h2>
        <ListCategories
          data={categories}
          onSearch={handleSearch}
          className={classes.listCategories}
        />
        <ListNews
          totalNews={totalNews}
          data={newsByCategory}
          categories={categories}
          className={classes.listNews}
        />
        <ListRankings
          data={newsRankings}
          categories={categories}
          className={classes.listRankings}
        />
      </Content>
      <Footer />
    </Layout>
  );
};
const newsPropTypes = PropTypes.arrayOf(
  PropTypes.shape({
    id: PropTypes.string.isRequired,
    thumbnail: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
  }),
);
NewsAndMedia.propTypes = {
  totalNews: PropTypes.number,
  categories: PropTypes.arrayOf(
    PropTypes.shape({ name: PropTypes.string.isRequired }),
  ),
  topNews: newsPropTypes,
  newsByCategory: newsPropTypes,
  newsRankings: newsPropTypes,
};
NewsAndMedia.defaultProps = {
  totalNews: 0,
  topNews: [],
  categories: [],
  newsRankings: [],
  newsByCategory: [],
};

export default memo(NewsAndMedia);
