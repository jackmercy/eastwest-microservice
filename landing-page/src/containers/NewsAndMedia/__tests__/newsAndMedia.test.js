import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import preloadAll from 'jest-next-dynamic';

import { Footer, Header } from '../../components';
import NewsAndMedia from '../NewsAndMedia';
import Outstanding from '../Outstanding';
import ListNews from '../ListNews';
import ListCategories from '../ListCategories';
import ListRankings from '../ListRankings';

describe('Testing NewsAndMedia', () => {
  let wrapper;
  const props = {
    totalNews: 10,
    categories: [
      {
        tags: ['Insights'],
        name: 'Insights',
        slug: 'insights',
        shortDescription: 'Insights',
        id: '610752dd1f174b00188e6359',
      },
    ],
    topNews: [
      {
        tags: [],
        title: 'title 1',
        slug: 'test-new-news',
        category: '610752dd1f174b00188e6359',
        author: '610752fc1f174b00188e635a',
        shortDescription: 'short description',
        htmlContent: '<p>Test 1</p>',
        thumbnail: '61075fb6c8e8db00182e1819',
        id: '61075fb61f174b00188e635b',
      },
    ],
    newsByCategory: [
      {
        tags: [],
        title: 'title 1',
        slug: 'test-new-news',
        category: '610752dd1f174b00188e6359',
        author: '610752fc1f174b00188e635a',
        shortDescription: 'short description',
        htmlContent: '<p>Test 1</p>',
        thumbnail: '61075fb6c8e8db00182e1819',
        id: '61075fb61f174b00188e635b',
      },
    ],
    newsRankings: [
      {
        tags: [],
        title: 'title 1',
        slug: 'test-new-news',
        category: '610752dd1f174b00188e6359',
        author: '610752fc1f174b00188e635a',
        shortDescription: 'short description',
        htmlContent: '<p>Test 1</p>',
        thumbnail: '61075fb6c8e8db00182e1819',
        id: '61075fb61f174b00188e635b',
      },
    ],
  };

  beforeAll(async () => {
    // Header and Footer included in Careers
    // And they are using dynamic import, see more in Header.js and Footer.js
    // So we need wait for it to load all before we can assert anything
    await preloadAll();
    wrapper = mount(<NewsAndMedia {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<NewsAndMedia {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a Header', () => {
    expect(wrapper.find(Header)).toHaveLength(1);
  });

  it('should have a Footer', () => {
    expect(wrapper.find(Footer)).toHaveLength(1);
  });

  it('should have Outstanding section', () => {
    expect(wrapper.find(Outstanding)).toHaveLength(1);
  });

  it('should have ListCategories section', () => {
    expect(wrapper.find(ListCategories)).toHaveLength(1);
  });

  it('should have ListNews section', () => {
    expect(wrapper.find(ListNews)).toHaveLength(1);
  });

  it('should have ListRankings section', () => {
    expect(wrapper.find(ListRankings)).toHaveLength(1);
  });
});
