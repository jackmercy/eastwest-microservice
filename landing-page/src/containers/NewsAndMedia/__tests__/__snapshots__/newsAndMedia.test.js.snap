// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NewsAndMedia should render match snapshot 1`] = `
<Memo(Layout)>
  <SEO
    creator=""
    description="Latest, Insights, Tips & Tricks, Articles, Data Talk & More East West News!"
    image=""
    keywords="East West, XP, News, Engineering, Product development, eXtreme Programming, UCD, User-centered design, Tech News, Lean Product Management, Growth, Remote, Freelance, Vietnam, Ireland"
    siteName=""
    title="East West - News & Media"
    url="https://dev.eastwest.live/news-and-media"
  />
  <ForwardRef(LoadableComponent) />
  <Memo(Outstanding)
    categories={
      Array [
        Object {
          "id": "610752dd1f174b00188e6359",
          "name": "Insights",
          "shortDescription": "Insights",
          "slug": "insights",
          "tags": Array [
            "Insights",
          ],
        },
      ]
    }
    data={
      Array [
        Object {
          "author": "610752fc1f174b00188e635a",
          "category": "610752dd1f174b00188e6359",
          "htmlContent": "<p>Test 1</p>",
          "id": "61075fb61f174b00188e635b",
          "shortDescription": "short description",
          "slug": "test-new-news",
          "tags": Array [],
          "thumbnail": "61075fb6c8e8db00182e1819",
          "title": "title 1",
        },
      ]
    }
  />
  <Memo(Content)
    className="content-0-2-1"
  >
    <h2
      className="title-0-2-2"
    >
      News & Media
    </h2>
    <ForwardRef(LoadableComponent)
      className="listCategories-0-2-3"
      data={
        Array [
          Object {
            "id": "610752dd1f174b00188e6359",
            "name": "Insights",
            "shortDescription": "Insights",
            "slug": "insights",
            "tags": Array [
              "Insights",
            ],
          },
        ]
      }
      onSearch={[Function]}
    />
    <Memo(ListNews)
      categories={
        Array [
          Object {
            "id": "610752dd1f174b00188e6359",
            "name": "Insights",
            "shortDescription": "Insights",
            "slug": "insights",
            "tags": Array [
              "Insights",
            ],
          },
        ]
      }
      className="listNews-0-2-4"
      data={
        Array [
          Object {
            "author": "610752fc1f174b00188e635a",
            "category": "610752dd1f174b00188e6359",
            "htmlContent": "<p>Test 1</p>",
            "id": "61075fb61f174b00188e635b",
            "shortDescription": "short description",
            "slug": "test-new-news",
            "tags": Array [],
            "thumbnail": "61075fb6c8e8db00182e1819",
            "title": "title 1",
          },
        ]
      }
      totalNews={10}
    />
    <Memo(ListRankings)
      categories={
        Array [
          Object {
            "id": "610752dd1f174b00188e6359",
            "name": "Insights",
            "shortDescription": "Insights",
            "slug": "insights",
            "tags": Array [
              "Insights",
            ],
          },
        ]
      }
      className="listRankings-0-2-5"
      data={
        Array [
          Object {
            "author": "610752fc1f174b00188e635a",
            "category": "610752dd1f174b00188e6359",
            "htmlContent": "<p>Test 1</p>",
            "id": "61075fb61f174b00188e635b",
            "shortDescription": "short description",
            "slug": "test-new-news",
            "tags": Array [],
            "thumbnail": "61075fb6c8e8db00182e1819",
            "title": "title 1",
          },
        ]
      }
    />
  </Memo(Content)>
  <ForwardRef(LoadableComponent) />
</Memo(Layout)>
`;
